src/Root.tsx

```diff
@@ -24,23 +26,32 @@ const FullScreen = styled.div`
   max-height: 100vh;
 `;
 
-const RootClass = () => {
+const { store, persistor } = createStore();
+
+const RootComponent = () => {
   const Router: any = USE_HASH_ROUTER ? HashRouter : BrowserRouter;
+  return (
+    <ThemeProvider theme={theme}>
+      <Router>
+        <AppProviders>
+          <FullHeight>
+            <DevToolsManager />
+            <FullScreen id="ModalContainer">
+              <AppRoutes />
+            </FullScreen>
+          </FullHeight>
+        </AppProviders>
+      </Router>
+    </ThemeProvider>
+  );
+};
 
+const RootClass = () => {
   return (
     <Provider store={store}>
-      <ThemeProvider theme={theme}>
-        <Router>
-          <AppProviders>
-            <FullHeight>
-              <DevToolsManager />
-              <FullScreen id="ModalContainer">
-                <AppRoutes />
-              </FullScreen>
-            </FullHeight>
-          </AppProviders>
-        </Router>
-      </ThemeProvider>
+      <PersistGate persistor={persistor}>
+        {(isHydrated: boolean) => (isHydrated ? <RootComponent /> : <AppLoading />)}
```

Comment:
Isn't this missing the `<Layout></Layout>`?

Reply:
`Layout` is connected to Providers which are not available at this moment.
Replaced with a simple `Box` to center the `AppLoading`